validation/validation.go

```diff
@@ -40,16 +40,32 @@ import (
 // .my.name@myself.com - match not found - invalid
 // my.name@myself.com. - match not found - invalid
 var emailRegex = regexp.MustCompile(`^(?:[a-z0-9!#$%&'*+/=?^_` + "`" + `{|}~-]+(?:\.[a-z0-9-!#$%&'*+/=?^_` + "`" + `{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)$`)
+var onsEmailRegex = regexp.MustCompile(`^(?:[a-z0-9!#$%&'*+/=?^_` + "`" + `{|}~-]+(?:\.[a-z0-9-!#$%&'*+/=?^_` + "`" + `{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(:?ext\.)?ons.gov.uk$`)
+var (
+	minimumEmailLength, maximumEmailLength int = 3, 254
+)
 
 //IsEmailValid is a function to validate email addresses. Here an email is valid if it follows the standard rules for valid email addresses. The unit tests contain examples of what are considered valid or invalid email addresses. Here we do not validate on the domain name.
 func IsEmailValid(e string) bool {
-
-	minimumEmailLength := 3
-	maximumEmailLength := 254
-
-	if len(e) < minimumEmailLength && len(e) > maximumEmailLength {
+	if !emailLengthValid(len(e)) {
 		return false
 	}
 
 	return emailRegex.MatchString(e)
 }
+
+// ValidateONSEmail - validates email address for ons domain
+// valid if match found, else invalid
+func IsValidateONSEmail (e string) bool {
+	if !emailLengthValid(len(e)) {
```

Comment:
This is really small but should the function be isValidONSEmail or ValidateONSEmail?

Reply:
👍